### ISSUE TITLE
gPodder doesn't recognize already downloaded files

### ISSUE THREAD

**Issue by butrus** [gpodder/gpodder#778]
If I put files already downloaded "outside" gPodder into the appropriate directory, gPodder still tries to download them anew, even if the file-size and file-name match. With really big podcasts it is really annoying one cannot add already downloaded files (especially having slow connection).

**Comment by elelay** [comment]
Do you have the 'Rename after download extension' enabled? Maybe the filename is not what you expect. Otherwise it should work. External files are rescanned only at gPodder startup, maybe you didn't restart?

Here is the code doing external file scan: https://github.com/gpodder/gpodder/blob/master/src/gpodder/model.py#L853
Maybe you can add some debugging and see why your files are not picked (I did this once to debug youtube podcasts not picking external files).

**Comment by butrus** [comment]
No, I don't have that extension enabled. I see two files in that directory. (Episode) and (Episode).part, where (Episode) is the file I put manually there before letting gpodder to load/update the feed, (Episode).part is created by gpodder.

This is really strange...

**Comment by auouymous** [comment]
This extension adds debug items to channel and episode menus, and outputs information about them to the console. Does download_filename match the file on disk? And does download_folder match the folder you placed the file in?

http://borg.qzx.com/files/gpodder-debug-extension.zip

**Comment by butrus** [comment]
OK, I realized now that it works but only after I restarted gPodder. If I have gPodder running, put the files in the directory and let gPodder update the podcast and then let it download some episode: it will download the file anew. Feel free to update the bug to reflect this, my english is not that good to formulate this in a meaningful manner. Thank you!

**Comment by elelay** [comment]
My English might not be so good as well. I'm marking this as enhancement, but I'm not sure what already exists and how to make this happen...
